src/main/java/mate/academy/spring/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.spring.security;
+
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        try {
+            User user = userService.findByEmail(email);
+            UserBuilder builder =
+                    org.springframework.security.core.userdetails.User.withUsername(email);
+            builder.password(user.getPassword());
+            builder.roles(user.getRoles()
+                    .stream()
+                    .map(r -> r.getRoleName().name())
+                    .toArray(String[]::new));
+            return builder.build();
+        } catch (DataProcessingException e) {
```

Comment:
It's recommended to put in try block only a part of code that can generate exception

Reply:
done